call-home/src/bin/stats/exporter/events_collector.rs

```diff
@@ -108,6 +119,47 @@ impl StatsCollector {
         metric_family.extend(pools_deleted.collect().pop());
         metric_family
     }
+
+    fn nexus_metrics(&self, events: &EventSet) -> Vec<prometheus::proto::MetricFamily> {
+        let mut metric_family = Vec::new();
+        let nexus_created = match self.nexus.get_metric_with_label_values(&[CREATED]) {
+            Ok(nexus) => nexus,
+            Err(error) => {
+                error!(%error,"Error while creating metrics(nexus created) with label values: {}", CREATED );
+                return metric_family;
+            }
+        };
+        nexus_created.inc_by(events.nexus.nexus_created as f64);
+        let nexus_deleted = match self.nexus.get_metric_with_label_values(&[DELETED]) {
+            Ok(nexus) => nexus,
+            Err(error) => {
+                error!(%error,"Error while creating metrics(nexus deleted) with label values: {}", DELETED);
+                return metric_family;
+            }
+        };
+        nexus_deleted.inc_by(events.nexus.nexus_deleted as f64);
+        let rebuild_begun = match self.nexus.get_metric_with_label_values(&[REBUILD_BEGUN]) {
+            Ok(nexus) => nexus,
+            Err(error) => {
+                error!(%error,"Error while creating metrics(rebuild begun) with label values: {}", REBUILD_BEGUN);
+                return metric_family;
+            }
+        };
+        rebuild_begun.inc_by(events.nexus.rebuild_begun as f64);
+        let rebuild_ended = match self.nexus.get_metric_with_label_values(&[REBUILD_ENDED]) {
+            Ok(nexus) => nexus,
+            Err(error) => {
+                error!(%error,"Error while creating metrics(rebuild ended) with label values: {}", REBUILD_ENDED);
+                return metric_family;
+            }
+        };
+        rebuild_ended.inc_by(events.nexus.rebuild_ended as f64);
+        metric_family.extend(nexus_created.collect().pop());
```

Comment:
Seems like a funny syntax, why are we collecting an popping?

Reply:
The collect() here is collects the single metric as a vector. But I guess pop() is not required here. Will check.

Reply:
pop() is not need. Removed it.

Reply:
Before removing it, let's understand why it was there. Was this a 1-element vector?

Reply:
Yes. it is a single element vector.